### PR TITLE
Command and DeleteCommand Feature

### PR DESCRIPTION
**Pull request by PureUsagi**
Created Command abstract class for future Command classes and DeleteCommand to remove item from inventory.

### CODE REVIEW

File: src/main/java/seedu/binbash/command/DeleteCommand.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package seedu.binbash.command;
+
+import java.util.regex.Pattern;
+import seedu.binbash.ItemList;
+
+public class DeleteCommand extends Command {
+    public static final String COMMAND_WORD = "delete";
+    public static final Pattern COMMAND_FORMAT =
+            Pattern.compile("delete\\s(?<index>\\d+)");
```
Comment (YHWong20) [inline src/main/java/seedu/binbash/command/DeleteCommand.java:9]: Might need to re-confirm if we are deleting by index or by item name. The draft UG seems to use item name instead.

File: src/main/java/seedu/binbash/command/DeleteCommand.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package seedu.binbash.command;
+
+import java.util.regex.Pattern;
+import seedu.binbash.ItemList;
+
+public class DeleteCommand extends Command {
+    public static final String COMMAND_WORD = "delete";
```
Comment (YHWong20) [inline src/main/java/seedu/binbash/command/DeleteCommand.java:7]: String constant name in `Command.java` is `COMMAND_STRING`, but here it's `COMMAND_WORD`. 

Not sure if this is intentional. If it is, then this comment can be ignored.

File: src/main/java/seedu/binbash/ItemList.java [hunk 3]
```diff
@@ -21,6 +21,15 @@ public void addItem(Item item) {
         itemList.add(item);
     }
 
+    public String deleteItem(int index) {
+        Item tempItem = itemList.get(index - 1);
+        itemList.remove(index - 1);
```
Comment (YHWong20) [inline src/main/java/seedu/binbash/ItemList.java:26]: If I'm not wrong, the `.remove(int index)` method for a List will return us the removed element, so we can probably consider removing the previous line. 

i.e., we can probably merge the two lines into something like `Item removedItem = itemList.remove(index - 1);`?

**Review by YHWong20 — CHANGES_REQUESTED** [review]

File: src/main/java/seedu/binbash/command/DeleteCommand.java (hunk 1 above)
Comment (YHWong20) [reply src/main/java/seedu/binbash/command/DeleteCommand.java:9]: I think we can probably ignore this comment.

File: src/main/java/seedu/binbash/ItemList.java (hunk 3 above)
Comment (PureUsagi) [reply src/main/java/seedu/binbash/ItemList.java:26]: Okayy, will remove the line.

File: src/main/java/seedu/binbash/command/DeleteCommand.java (hunk 2 above)
Comment (PureUsagi) [reply src/main/java/seedu/binbash/command/DeleteCommand.java:7]: Oh yea, it's a typo. Forgot to change the variable name.

**Pull request closed without merging**